modules/sample/src/main/resources/validation/example.yaml

```diff
@@ -32,6 +32,9 @@ definitions:
         format: int32
         minimum: 0
         maximum: 100
+      pattern_validation:
+        type: string
+        pattern: "[0-9]+"
```

Comment:
One lingering issue. According to [the docs](https://swagger.io/docs/specification/data-models/data-types/#pattern)...

> Without `^…$`, pattern works as a partial match, that is, matches any string that contains the specified regular expression.

It looks as though this violates the assumptions baked into refined:

```scala
scala> val pat = _root_.shapeless.Witness("[0-9]+")
pat: shapeless.Witness.Aux[String("[0-9]+")] = shapeless.Witness$$anon$1@7d875a3e

scala> val x: String Refined MatchesRegex[pat.T] = "foo1234bar"
<console>:24: error: Predicate failed: "foo1234bar".matches("[0-9]+").
       val x: String Refined MatchesRegex[pat.T] = "foo1234bar"
                                                   ^
```

Before we merge, could you look into this please?

Reply:
that is a good catch, I'll prepend/append `.*` if there is no `^` or `$` accordingly